corehq/apps/es/utils.py

```diff
@@ -32,3 +35,61 @@ def flatten_field_dict(results, fields_property='fields'):
             new_val = val[0]
         field_dict[key] = new_val
     return field_dict
+
+
+def chunk_query(query, term, chunk_size=1000):
+    query = deepcopy(query)
+
+    path_to_term = _chunk_query([], None, query.es_query, term)
+    chunked_queries = []
+
+    # No path to term could be found, return just the query
+    if path_to_term is None:
+        return [query]
+
+    term_list = _dict_lookup(path_to_term, query.es_query)
+
+    # Cannot chunk term that is not a list
+    if not isinstance(term_list, list):
+        return [query]
```

Comment:
actually that ends up being kind of tricky because there are situations where the path to the term could have an `AND` clause in it. see the example in the test: https://github.com/dimagi/commcare-hq/pull/14677/files#diff-154c27feeb92507c0618c4e130dbb73dR17